src/screen/ConnectionsList/NewConnectionModal/index.tsx

```diff
@@ -62,12 +68,16 @@ const NewConnectionModal: React.FC<SharedModalProps> = ({
         const { name, host, port, password } = data
 
         try {
-          const connections = saveAndGetConnections({
+          const newConnectionData = {
```

Comment:
What about `connectionData` instead of `newConnectionData` as we are not always creating a new connection.